runtime/src/main/java/com/datastax/oss/driver/shaded/guava/common/primitives/UnsignedBytesSubstitutions.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.shaded.guava.common.primitives;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Delete;
+import com.oracle.svm.core.annotate.RecomputeFieldValue;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import java.util.Comparator;
+
+@TargetClass(UnsignedBytes.LexicographicalComparatorHolder.UnsafeComparator.class)
+@Delete
+final class UnsafeComparatorSubstitutions {}
+
+@TargetClass(UnsignedBytes.LexicographicalComparatorHolder.class)
+final class LexicographicalComparatorHolderSubstitutions {
+
+  @Alias
+  @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.FromAlias)
+  static Comparator<byte[]> BEST_COMPARATOR = UnsignedBytes.lexicographicalComparatorJavaImpl();
```

Comment:
Which class access was the main problem? Is it `ByteOrder` that uses `jdk.internal.misc.Unsafe`?

Reply:
They have an [impl](https://github.com/google/guava/blob/v25.1/guava/src/com/google/common/primitives/UnsignedBytes.java#L294-L409) of their byte array comparator which uses Unsafe for raw byte operations.  There's also a [pure Java](https://github.com/google/guava/blob/v25.1/guava/src/com/google/common/primitives/UnsignedBytes.java#L411-L430) version available as a fallback, but basically if you're on a 64-bit platform you're pretty likely to get the Unsafe-based version by default.  The [selection logic](https://github.com/google/guava/blob/v25.1/guava/src/com/google/common/primitives/UnsignedBytes.java#L432-L448) relies on a [static init op in the Unsafe impl](https://github.com/google/guava/blob/v25.1/guava/src/com/google/common/primitives/UnsignedBytes.java#L320-L329) to determine whether it's safe to use.  It's the Unsafe.arrayIndexScale() op in that static init op which is the root problem here.  Because that assignment isn't written to a field Graal can't re-write it's value from the build-time architecture (where it's actually computed) to Graal's architecture (which is the value it would need at runtime).  This generates the warnings.